config/configauth/serverauth.go

```diff
@@ -40,7 +40,7 @@ type ServerAuthenticator interface {
 	// When the authentication fails, an error must be returned and the caller must not retry. This function is typically called from interceptors,
 	// on behalf of receivers, but receivers can still call this directly if the usage of interceptors isn't suitable.
 	// The deadline and cancellation given to this function must be respected, but note that authentication data has to be part of the map, not context.
-	Authenticate(ctx context.Context, headers map[string][]string) error
+	Authenticate(ctx context.Context, headers map[string][]string) (context.Context, error)
```

Comment:
Can we add an explanation in the comments about what the returned context is expected to be?

Reply:
I left that part out so that we can define it when we have a first implementation making use of it, but I added the best information we have so far.

Should I define the context keys as part of this PR? It would be basically this: https://github.com/open-telemetry/opentelemetry-collector/pull/3239/files#diff-1c8371e19129e2092ab7abfce04e6558218579f1be29d90adb958d230475584f

Reply:
In any case, I updated the PR to include a few words about the resulting context.

Reply:
Sorry about the noise, but I thought it would be better to include the auth keys as part of the API for v1.